### DISCUSSION TITLE
Injecting span context using SpanContext instead of Context

### DISCUSSION THREAD

**Discussion by Deeds67** [open-telemetry/opentelemetry-java#4456] [Q&A] [answered]
Hi, the old `opentracing-api`'s `inject` signature supported injecting a `SpanContext`:

```
<C> void inject(SpanContext spanContext, Format<C> format, C carrier);
```

The new signature only takes a `Context`:

```
<C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> setter)
```

Since I'm passing SpanContext on MDC (Using Scala with monix task and monix-mdc. I don't think thread based context propagation will work), I'm not using `Context` at all, but it seems like the new opentelemetry API doesn't allow me to inject span context using `SpanContext` anymore. Is there a plan to support this?

**Comment by jkwatson** [comment, marked as answer]
You can put your SpanContext into a Context today like this (perhaps not the most obvious API, we know):
```
context.with(Span.wrap(spanContext))
```

**Reply by Deeds67** [reply]
Thanks for the response. 

Is it possible to avoid `Context` completely? I'm manually passing the `SpanContext` because `Context` looks like it uses ThreadLocal and I think Monix Task doesn't support this out of the box. At least I haven't investigated much into why it's not working, but I prefer as an initial approach to just manually pass the `SpanContext`. The old opentracing api allowed this, is there still a way to do this?

**Reply by jkwatson** [reply]
In OpenTelemetry, Context is the unit of propagation, not SpanContext, so you can't avoid creating a Context instance to do this, no. If you never make your Context "current", then no ThreadLocal will be involved.

**Reply by Deeds67** [reply]
Thank you very much for the assistance. This has solved my problem 👍
